### PR TITLE
supporting auto subscribers (DEV-7015, DEV-7101)

### PR DESCRIPTION
**Pull request by peter-horvath**

### CODE REVIEW

File: src/Toggle/StatelessToggleChecker.php [hunk 1]
```diff
@@ -74,6 +87,16 @@ private function isGroupToggleInUnsetGroup(UseCase\Response\TogglePolicyResponse
...
+        if ( !$isGroupPolicyNotSet ) {
+            return false;
+        }
+
+        $isUserAutoSubscriber = $this->autoSubscribersGateway->isUserAutoSubscriber( $user );
+        $userTogglePolicy = $this->userPolicy->getTogglePolicy($toggleName, $user);
+        $isUserPolicyNotSet = $userTogglePolicy->isNotSet();
+        $isUserPolicyEnabled = $userTogglePolicy->isEnabled();
+
+        return ( $isUserAutoSubscriber && $isUserPolicyNotSet ) || $isUserPolicyEnabled;
```
Comment (adrianclay) [inline src/Toggle/StatelessToggleChecker.php]: Looks like ``` $isUserPolicyNotSet ``` will always be true?

File: test/Toggle/IsCurrentUserToggleActiveTest.php [hunk 2]
```diff
@@ -121,6 +136,65 @@ public function WhenToggleInvisibleAndNotEnabledByPolicyButReleaseDateIsTodayOrI
     }
 
     /**
+     * @test
+     */
+    public function GivenToggleInvisibleAndNotSetByPolicyAndReleaseDateIsInTheFutureAndUserIsAutoSubscribed_ThenInactive()
+    {
+        $this->setupChecker( false, false, false, false, false, false, false );
```
Comment (adrianclay) [inline test/Toggle/IsCurrentUserToggleActiveTest.php]: I'm finding this setup method quite unreadable.

This is how I would create a similar but more readable arrange:

``` php

public function setUp() {
  parent::setUp();
  $this->isAdrianCommenting = true;
  $this->numberOfDaysInMonth = 30;
  $this->toggleVisibility = 0;
}

public function givenFebruaryAndVisibleToggle_ThenIsSpooky()
{
  $this->numberOfDaysInMonth = 28;
  $this->toggleVisibility = 100;
  $this->act();
  $this->assertSpooky();
}
```

Comment (peter-horvath) [inline test/Toggle/IsCurrentUserToggleActiveTest.php]: Yeah I don't like it either at all :S I just didn't want to refactor all the tests and I saw all of them were using this, so I kept using it too. Should I refactor the whole file?

Comment (adrianclay) [inline test/Toggle/IsCurrentUserToggleActiveTest.php]: Do you think you could do it in 15 minutes?

Comment (peter-horvath) [inline test/Toggle/IsCurrentUserToggleActiveTest.php]: I'm not 100% sure, but probably it wouldn't take more than an hour. I'll take a look at this tomorrow.

**Comment by peter-horvath** [comment]
@adrianclay I refactored those tests, what do you think now?

**Comment by adrianclay** [comment]
I'm thinking this is top drawer.

**Comment by peter-horvath** [comment]
Cheers :)

**Pull request merged**